### PR TITLE
feat(suite): add headingSize props to Modal

### LINKED ISSUE #10052: Add modal heading size prop to Modal
**Design:** [Figma](https://www.figma.com/file/Y2E8dcdNYrSMGIVRjYFXKp/%5Bs2.0%5D-Components?type=design&node-id=57-31270&mode=dev)

* add the `headingSize` prop, only have `default` and `large` for now, since they are the ones being used.
* use `large` for the Wallet Switch and New Account modals

### PR DESCRIPTION
**Pull request by peter-sanderson**
## Description

Add default and large font-size for the Modals Heading

- The size of heading for Wallet Switch and New Account modals increased to `larger`
- On top, I have also increased the `AddAccountModal` as it seems to me as same type of modal

## Related Issue

Resolve https://github.com/trezor/trezor-suite/issues/10052

## Screenshots:

image
image

image
image

image
image

### CODE REVIEW

File: packages/components/src/components/modals/Modal/Modal.tsx [hunk 1]
```diff
@@ -79,7 +80,14 @@ const HeadingContainer = styled.div<HeadingContainerProps>`
         `}
 `;
 
-const Heading = styled(H3)<{ isWithIcon?: boolean }>`
+const HEADING_SIZES = {
```
Comment (dahaca) [inline packages/components/src/components/modals/Modal/Modal.tsx]:
```suggestion
const HEADING_SIZES: Record<
    HeadingSize,
    {
        css: string;
        buttonSize: ButtonSize;
    }
> = {
```

File: packages/components/src/components/modals/Modal/Modal.tsx [hunk 2]
```diff
@@ -79,7 +80,14 @@ const HeadingContainer = styled.div<HeadingContainerProps>`
...
+    default: { css: typography.titleSmall, buttonSize: 'small' as ButtonSize },
+    large: { css: typography.titleMedium, buttonSize: 'medium' as ButtonSize },
+};
+
+type HeadingProps = { isWithIcon?: boolean; $headingSize?: keyof typeof HEADING_SIZES };
```
Comment (dahaca) [inline packages/components/src/components/modals/Modal/Modal.tsx]:
```suggestion
type HeadingProps = { isWithIcon?: boolean; $headingSize?: HeadingSize };
```

File: packages/components/src/components/modals/Modal/Modal.tsx [hunk 3]
```diff
@@ -181,6 +191,7 @@ const Modal = ({
     subheading,
     modalPrompt,
     headerIcon,
+    headingSize,
```
Comment (dahaca) [inline packages/components/src/components/modals/Modal/Modal.tsx]: So you don't need the `??` later

```suggestion
    headingSize = 'default',
```

File: packages/components/src/components/modals/Modal/Modal.tsx [hunk 4]
```diff
@@ -161,6 +170,7 @@ interface ModalProps {
     subheading?: ReactNode;
     modalPrompt?: ReactNode;
     headerIcon?: IconType;
+    headingSize?: keyof typeof HEADING_SIZES;
```
Comment (dahaca) [inline packages/components/src/components/modals/Modal/Modal.tsx]:
```suggestion
    headingSize?: HeadingSize;
```

**Comment by komret** [comment]
No need to change it now, but for next time I'd suggest splitting it into two commits - one for the change in `@trezor/components`, one for implementing it in `@trezor/suite`.

File: packages/suite/src/views/suite/SwitchDevice/SwitchDevice.tsx [hunk 5]
```diff
@@ -41,6 +41,7 @@ export const SwitchDevice = ({ cancelable, onCancel }: ForegroundAppProps) => {
 
     return (
         <Modal
+            headingSize="large"
```
Comment (komret) [inline packages/suite/src/views/suite/SwitchDevice/SwitchDevice.tsx]: I misguided you about this modal, sorry, it should be `default`, see https://www.figma.com/file/mQhdQPKzdi93Swb8FNX9wL/branch/pewjlOfequOKYVC6h3v0YL/%5Bs2.0%5D-Desktop?type=design&node-id=3276%3A171291&mode=dev

File: packages/components/src/components/modals/Modal/Modal.tsx (hunk 1 above)
Comment (peter-sanderson) [reply packages/components/src/components/modals/Modal/Modal.tsx]: We agreed with @dahaca to not doing this.

File: packages/components/src/components/modals/Modal/Modal.tsx (hunk 4 above)
Comment (peter-sanderson) [reply packages/components/src/components/modals/Modal/Modal.tsx]: :heavy_check_mark:

File: packages/components/src/components/modals/Modal/Modal.tsx (hunk 3 above)
Comment (peter-sanderson) [reply packages/components/src/components/modals/Modal/Modal.tsx]: :heavy_check_mark:

File: packages/suite/src/views/suite/SwitchDevice/SwitchDevice.tsx (hunk 5 above)
Comment (peter-sanderson) [reply packages/suite/src/views/suite/SwitchDevice/SwitchDevice.tsx]: :heavy_check_mark:

File: packages/components/src/components/modals/Modal/Modal.tsx (hunk 2 above)
Comment (peter-sanderson) [reply packages/components/src/components/modals/Modal/Modal.tsx]: :heavy_check_mark:

File: packages/components/src/components/modals/Modal/Modal.tsx [hunk 6]
```diff
@@ -97,7 +99,7 @@ const Heading = styled(H3)<HeadingProps>`
                 margin-right: ${spacingsPx.xs};
             }
         `}
-    ${({ $headingSize }) => HEADING_SIZES[$headingSize ?? 'default'].css};
+    ${({ $headingSize = 'default' }) => HEADING_SIZES[$headingSize].css};
```
Comment (komret) [inline packages/components/src/components/modals/Modal/Modal.tsx]: Nit but you can remove the default value here if you set `$headingSize` as required.

**Review by komret — APPROVED** [review]
cc @dahaca https://github.com/trezor/trezor-suite/pull/10183 broke the validation check. I don't know what exactly it was. Fixed by Prettier auto-formatting in fcf67dd4ea583782913bd10db8b3c8a7538d5b3f
Screenshot 2023-12-07 at 19 07 52

Anyways, this PR is good to go! 👏

File: packages/suite/src/views/suite/SwitchDevice/SwitchDevice.tsx (hunk 5 above)
Comment (dahaca) [reply packages/suite/src/views/suite/SwitchDevice/SwitchDevice.tsx]: Wallets is Title/Medium ;)

Comment (komret) [reply packages/suite/src/views/suite/SwitchDevice/SwitchDevice.tsx]: > use large for the Wallet Switch and New Account modals

Hmmm so which modal do you mean by Wallet Switch, @dahaca?

Comment (dahaca) [reply packages/suite/src/views/suite/SwitchDevice/SwitchDevice.tsx]: this

image

Comment (komret) [reply packages/suite/src/views/suite/SwitchDevice/SwitchDevice.tsx]: So it seems to be correct, Wallets is set to `titleMedium` here, I am talking about the `SwitchDevice` modal above, but this is the `PassphraseModal`... it is quite confusing. But I believe it is correct now.

**Pull request merged**